utils/dwarf.c

```diff
@@ -1054,9 +1153,26 @@ static void build_dwarf_info(struct debug_info *dinfo, struct symtab *symtab,
 
 		dwarf_getfuncs(&cudie, get_dwarfspecs_cb, &bd, 0);
 
+		if (dwarf_hasattr(&cudie, DW_AT_comp_dir)) {
+			comp_dir = str_attr(&cudie, DW_AT_comp_dir, false);
+			add_dir_entry(&dirs, comp_dir, dinfo->nr_locs_used);
```

Comment:
Please copy the string here and free it in `free_dir_entry` so that the ownership can be explicit.

Reply:
ok. thank you. but I copied the str in add_comp_dir to make pair with free_comp_dir as you did in add_debug_entry. please let me know if you have another opinion. thank you.